prohibited_super_call rule false violation with NSFileProviderExtension providePlaceholder override

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

The prohibited_super_call rule generates an incorrect violation report when linting the `providePlaceholder` override for an `NSFileProviderExtension` implementation. The code to repro shown below is taken directly from the Apple template code generated for a new FileProvider extension target and does not call super.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint --config swiftlint.yml 
Loading configuration from 'swiftlint.yml'
Linting Swift files in current working directory
Linting 'FileProviderExtension.swift' (1/1)
/Users/bestave/Desktop/DeleteMe/DeleteMeFP/ProhibitedSuperTest/FileProviderExtension.swift:5:99: warning: Prohibited calls to super Violation: Method 'providePlaceholder(at:completionHandler:)' should not call to super function (prohibited_super_call)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version: 0.25.1 (release and latest source)
* Installation method used: SwiftLint.pkg installer and also cloned/built from master source
* Paste your configuration file:

```yml
opt_in_rules:
  - prohibited_super_call
```

* Are you using [nested configurations]? No
* Which Xcode version are you using? 9.3.1
* Do you have a sample that shows the issue? Yes:

```swift
import FileProvider

public class FileProviderExtension: NSFileProviderExtension {

    override func providePlaceholder(at url: URL, completionHandler: @escaping (Error?) -> Void) {
        guard let identifier = persistentIdentifierForItem(at: url) else {
            completionHandler(NSFileProviderError(.noSuchItem))
            return
        }
    }

}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -106,6 +106,10 @@
   [Marcelo Fabri](https://github.com/marcelofabri)
   [#2211](https://github.com/realm/SwiftLint/issues/2211)
 
+* Fix false positives in `prohibited_super_call` rule.  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#2212](https://github.com/realm/SwiftLint/issues/2212)
+
 ## 0.25.1: Lid Locked
 
 This is the last release to support building with Swift 3.2 and Swift 3.3.

--- a/Rules.md
+++ b/Rules.md
@@ -11890,18 +11890,28 @@ Some methods should not call super
 
 ```swift
 class VC: UIViewController {
-	override func loadView() {
-	}
+    override func loadView() {
+    }
 }
-
 ```
 
 ```swift
 class NSView {
-	func updateLayer() {
-		self.method1()	}
+    func updateLayer() {
+        self.method1()
+    }
 }
+```
 
+```swift
+public class FileProviderExtension: NSFileProviderExtension {
+    override func providePlaceholder(at url: URL, completionHandler: @escaping (Error?) -> Void) {
+        guard let identifier = persistentIdentifierForItem(at: url) else {
+            completionHandler(NSFileProviderError(.noSuchItem))
+            return
+        }
+    }
+}
 ```
 
 </details>
@@ -11910,43 +11920,39 @@ class NSView {
 
 ```swift
 class VC: UIViewController {
-	override func loadView() {↓
-		super.loadView()
-	}
+    override func loadView() {↓
+        super.loadView()
+    }
 }
-
 ```
 
 ```swift
 class VC: NSFileProviderExtension {
-	override func providePlaceholder(at url: URL,completionHandler: @escaping (Error?) -> Void) {↓
-		self.method1()
-		super.providePlaceholder(at:url, completionHandler: completionHandler)
-	}
+    override func providePlaceholder(at url: URL, completionHandler: @escaping (Error?) -> Void) {↓
+        self.method1()
+        super.providePlaceholder(at:url, completionHandler: completionHandler)
+    }
 }
-
 ```
 
 ```swift
 class VC: NSView {
-	override func updateLayer() {↓
-		self.method1()
-		super.updateLayer()
-		self.method2()
-	}
+    override func updateLayer() {↓
+        self.method1()
+        super.updateLayer()
+        self.method2()
+    }
 }
-
 ```
 
 ```swift
 class VC: NSView {
-	override func updateLayer() {↓
-		defer {
-			super.updateLayer()
-		}
-	}
+    override func updateLayer() {↓
+        defer {
+            super.updateLayer()
+        }
+    }
 }
-
 ```
 
 </details>

--- a/Source/SwiftLintFramework/Extensions/Dictionary+SwiftLint.swift
+++ b/Source/SwiftLintFramework/Extensions/Dictionary+SwiftLint.swift
@@ -122,11 +122,14 @@ extension Dictionary where Key: ExpressibleByStringLiteral {
     }
 
     internal func extractCallsToSuper(methodName: String) -> [String] {
-        let superCall = "super.\(methodName)"
+        guard let methodNameWithoutArguments = methodName.split(separator: "(").first else {
+            return []
+        }
+        let superCall = "super.\(methodNameWithoutArguments)"
         return substructure.flatMap { elems -> [String] in
-            guard let type = elems.kind.flatMap({ SwiftExpressionKind(rawValue: $0) }),
+            guard let type = elems.kind.flatMap(SwiftExpressionKind.init),
                 let name = elems.name,
-                type == .call && superCall.contains(name) else {
+                type == .call && superCall == name else {
                     return elems.extractCallsToSuper(methodName: methodName)
             }
             return [name]

--- a/Source/SwiftLintFramework/Rules/ProhibitedSuperRule.swift
+++ b/Source/SwiftLintFramework/Rules/ProhibitedSuperRule.swift
@@ -11,43 +11,64 @@ public struct ProhibitedSuperRule: ConfigurationProviderRule, ASTRule, OptInRule
         description: "Some methods should not call super",
         kind: .lint,
         nonTriggeringExamples: [
-            "class VC: UIViewController {\n" +
-                "\toverride func loadView() {\n" +
-                "\t}\n" +
-            "}\n",
-            "class NSView {\n" +
-                "\tfunc updateLayer() {\n" +
-                    "\t\tself.method1()" +
-                "\t}\n" +
-            "}\n"
+            """
+            class VC: UIViewController {
+                override func loadView() {
+                }
+            }
+            """,
+            """
+            class NSView {
+                func updateLayer() {
+                    self.method1()
+                }
+            }
+            """,
+            """
+            public class FileProviderExtension: NSFileProviderExtension {
+                override func providePlaceholder(at url: URL, completionHandler: @escaping (Error?) -> Void) {
+                    guard let identifier = persistentIdentifierForItem(at: url) else {
+                        completionHandler(NSFileProviderError(.noSuchItem))
+                        return
+                    }
+                }
+            }
+            """
         ],
         triggeringExamples: [
-            "class VC: UIViewController {\n" +
-                "\toverride func loadView() {↓\n" +
-                    "\t\tsuper.loadView()\n" +
-                "\t}\n" +
-            "}\n",
-            "class VC: NSFileProviderExtension {\n" +
-                "\toverride func providePlaceholder(at url: URL," +
-                "completionHandler: @escaping (Error?) -> Void) {↓\n" +
-                    "\t\tself.method1()\n" +
-                    "\t\tsuper.providePlaceholder(at:url, completionHandler: completionHandler)\n" +
-                "\t}\n" +
-            "}\n",
-            "class VC: NSView {\n" +
-                "\toverride func updateLayer() {↓\n" +
-                    "\t\tself.method1()\n" +
-                    "\t\tsuper.updateLayer()\n" +
-                    "\t\tself.method2()\n" +
-                "\t}\n" +
-            "}\n",
-            "class VC: NSView {\n" +
-                "\toverride func updateLayer() {↓\n" +
-                "\t\tdefer {\n" +
-                "\t\t\tsuper.updateLayer()\n" +
-                "\t\t}\n" +
-                "\t}\n" +
-            "}\n"
+            """
+            class VC: UIViewController {
+                override func loadView() {↓
+                    super.loadView()
+                }
+            }
+            """,
+            """
+            class VC: NSFileProviderExtension {
+                override func providePlaceholder(at url: URL, completionHandler: @escaping (Error?) -> Void) {↓
+                    self.method1()
+                    super.providePlaceholder(at:url, completionHandler: completionHandler)
+                }
+            }
+            """,
+            """
+            class VC: NSView {
+                override func updateLayer() {↓
+                    self.method1()
+                    super.updateLayer()
+                    self.method2()
+                }
+            }
+            """,
+            """
+            class VC: NSView {
+                override func updateLayer() {↓
+                    defer {
+                        super.updateLayer()
+                    }
+                }
+            }
+            """
         ]
     )
 

